geoapps/interpolation/driver.py

```diff
@@ -0,0 +1,295 @@
+#  Copyright (c) 2022 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+
+from __future__ import annotations
+
+import sys
+from os import path
+
+import numpy as np
+from discretize.utils import mesh_utils
+from geoh5py.objects import BlockModel, ObjectBase
+from geoh5py.ui_json import InputFile
+from geoh5py.ui_json.utils import monitored_directory_copy
+from scipy.interpolate import LinearNDInterpolator
+from scipy.spatial import cKDTree
+
+from geoapps.interpolation.params import DataInterpolationParams
+from geoapps.shared_utils.utils import get_locations, weighted_average
+
+
+class DataInterpolationDriver:
+    def __init__(self, params: DataInterpolationParams):
+        self.params: DataInterpolationParams = params
+        self._unique_object = {}
+        self.object_out = None
+
+    def object_base(self, object):
```

Comment:
I like the idea of specifying a type on the workspace.get_entity(). I don't think it is necessary here though since the ui.json already enforces object types on selection.